pkg/webhook/failed_status_checks_add_comment.go

```diff
@@ -54,18 +54,41 @@ func (h *failedStatusCheckAddComment) HandleEvent(w http.ResponseWriter, eventOb
 	commitSHA := event.GetSHA()
 	query := fmt.Sprintf("type:pr state:open repo:%s %s", event.Repo.GetFullName(), commitSHA)
 	searchResult, _, err := gh.Search.Issues(context.Background(), query, nil)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find PR using query %s", query)
+	}
+
+	owner, repo := event.Repo.Owner.GetLogin(), event.Repo.GetName()
+
 	var multiErr error
 	for _, issue := range searchResult.Issues {
 		if issue.PullRequestLinks == nil {
 			continue
 		}
 
-		msg := fmt.Sprintf("Status check _%s_ returned **%s**.", event.GetContext(), state)
+		prNumber, prURL := issue.GetNumber(), issue.GetHTMLURL()
+
+		message := fmt.Sprintf("Status check _%s_ returned **%s**.", event.GetContext(), state)
 		if event.GetTargetURL() != "" {
-			msg += fmt.Sprintf(" See %s for more details.", event.GetTargetURL())
+			message += fmt.Sprintf(" See %s for more details.", event.GetTargetURL())
+		}
+
+		existingComments, _, err := gh.Issues.ListComments(context.Background(), owner, repo, prNumber, &github.IssueListCommentsOptions{
+			Sort:      "updated",
+			Direction: "desc",
+		})
+		if err != nil {
+			return errors.Wrapf(err, "failed to retrieve existing comments on PR %s", prURL)
+		}
+
+		for _, existingComment := range existingComments {
+			if message == existingComment.GetBody() {
```

Comment:
The problem isn't only the same exact event being redelivered, it's multiple distinct events for same status check. I saw earlier that from a circleci status check. So this is more about trying to ensure there are not multiple exact same messages on a PR. It is expected that the status check target URL should make the message distinct enough, but of course that is not guaranteed - although for everything we're currently using it is enough...